misk-hibernate/src/main/kotlin/misk/jdbc/DataSourceService.kt

```diff
@@ -22,7 +24,8 @@ internal class DataSourceService(
   private val qualifier: KClass<out Annotation>,
   private val config: DataSourceConfig,
   private val environment: Environment,
-  private val dataSourceDecorators: Set<DataSourceDecorator>
+  private val dataSourceDecorators: Set<DataSourceDecorator>,
+  private val collectorRegistry: CollectorRegistry? = null
```

Comment:
Why is this nullable?

Reply:
I think because not every client of misk-hibernate will have metrics installed?

Reply:
This change requires all usages of HibernateModule to have metrics installed, since the binding is here https://github.com/square/misk/pull/815/files#diff-212704a7927dd09259da3fe20775f0b0R96. It's not optional.

Reply:
Actually yeah, this is a breaking change for HibernateModule. @jmuia you might want to make adding the registry a `DataSourceDecorator` bound in a different module instead.

Reply:
rather surprising that the tests go green

Reply:
Hmm I think by default you are getting an implicit binding with a new instance of CollectorRegistry, which is almost certainly not what should get injected.

Reply:
I'll take a look at `DataSourceDecorator`. Misk installs metrics by default in `MiskCommonServiceModule`, which is why the tests are ok.

I made it nullable because some tests were creating the DataSourceService manually, but I can refactor a bit and have them pass in the CollectorRegistry.

Reply:
`CollectorRegistry` is explicitly bound in `MetricsModule` as a singleton.

Reply:
Yes, but if you don't install `MetricsModule()`, then someone using `HibernateModule()` but not `MetricsModule()` will get an implicit binding which will construct a new instance. When #816 lands, it'll be an error I have to fix...